### PR TITLE
NETOBSERV-1965: packet translation feature support

### PR DESCRIPTION
**Pull request by msherif1234**
Signed-off-by: Mohamed Mahmoud <mmahmoud@redhat.com>

## Dependencies

n/a

## Checklist

If you are not familiar with our processes or don't know what to answer in the list below, let us know in a comment: the maintainers will take care of that.

* [ ] Is this PR backed with a JIRA ticket? If so, make sure it is written as a title prefix _(in general, PRs affecting the NetObserv/Network Observability product should be backed with a JIRA ticket - especially if they bring user facing changes)._
* [ ] Does this PR require product documentation?
  * [ ] If so, make sure the JIRA epic is labelled with "documentation" and provides a description relevant for doc writers, such as use cases or scenarios. Any required step to activate or configure the feature should be documented there, such as new CRD knobs.
* [ ] Does this PR require a product release notes entry?
  * [ ] If so, fill in "Release Note Text" in the JIRA.
* [ ] Is there anything else the QE team should know before testing? E.g: configuration changes, environment setup, etc.
  * [ ] If so, make sure it is described in the JIRA ticket.
* QE requirements (check 1 from the list):
  * [ ] Standard QE validation, with pre-merge tests unless stated otherwise.
  * [ ] Regression tests only (e.g. refactoring with no user-facing change).
  * [ ] No QE (e.g. trivial change with high reviewer's confidence, or per agreement with the QE team).

### CODE REVIEW

**Comment by msherif1234** [comment]
flow collector configs
```yaml
type: eBPF
    ebpf:
      sampling: 1
      privileged: true
      features:
      - "PacketTranslation"
```

**Comment by msherif1234** [comment]
image

File: controllers/consoleplugin/config/static-frontend-config.yaml [hunk 1]
```diff
@@ -526,6 +526,78 @@ columns:
...
+    default: false
+    width: 10
+    feature: packetTranslation
+  - id: XlatSrcK8S_Object
+    group: Xlat
+    name: Xlat Src Kubernetes Object
+    calculated: kubeObject(SrcK8S_Type,SrcK8S_Namespace,SrcK8S_Name,1) or concat(XlatSrcAddr,':',XlatSrcPort)
+    default: false
+    width: 15
+    feature: packetTranslation
```
Comment (jotak) [inline controllers/consoleplugin/config/static-frontend-config.yaml:568]: can you remove it from the time being?
as discussed, this needs changes in FLP which aren't done yet
I'll add that to the [follow-up](https://issues.redhat.com/browse/NETOBSERV-2014)

File: controllers/consoleplugin/config/static-frontend-config.yaml [hunk 2]
```diff
@@ -526,6 +526,78 @@ columns:
...
+  - id: XlatK8S_Object
+    group: Xlat
+    name: Xlat Kubernetes Objects
+    calculated: '[column.XlatSrcK8S_Object,column.XlatDstK8S_Object]'
+    default: false
+    width: 15
+    feature: packetTranslation
```
Comment (jotak) [inline controllers/consoleplugin/config/static-frontend-config.yaml:600]: same here, to remove

File: controllers/flp/flp_pipeline_builder.go [hunk 3]
```diff
@@ -89,6 +89,22 @@ func (b *PipelineBuilder) AddProcessorStages() error {
...
+			},
+		},
+		{
+			Type: api.NetworkAddKubernetes,
+			Kubernetes: &api.K8sRule{
+				IPField: "XlatDstAddr",
+				Output:  "XlatDstK8S",
+				AddZone: addZone,
+			},
+		},
```
Comment (jotak) [inline controllers/flp/flp_pipeline_builder.go:107]: Zone (which is about cloud AZ) is defined out of the main IPs not the xlated ones.

```suggestion
		{
			Type: api.NetworkAddKubernetes,
			Kubernetes: &api.K8sRule{
				IPField: "XlatSrcAddr",
				Output:  "XlatSrcK8S",
			},
		},
		{
			Type: api.NetworkAddKubernetes,
			Kubernetes: &api.K8sRule{
				IPField: "XlatDstAddr",
				Output:  "XlatDstK8S",
			},
		},
```

File: controllers/consoleplugin/config/static-frontend-config.yaml [hunk 4]
```diff
@@ -526,6 +526,78 @@ columns:
...
+    tooltip: The packet translation source port number.
+    field: XlatSrcPort
+    filter: xlat_src_port
+    default: false
+    width: 10
+    feature: packetTranslation
+  - id: XlatSrcK8S_Object
+    group: Xlat
+    name: Xlat Src Kubernetes Object
+    calculated: kubeObject(SrcK8S_Type,SrcK8S_Namespace,SrcK8S_Name,1) or concat(XlatSrcAddr,':',XlatSrcPort)
```
Comment (jotak) [inline controllers/consoleplugin/config/static-frontend-config.yaml:565]: Based on how you configured FLP below, I guess it should be this: 

```suggestion
    calculated: kubeObject(XlatSrcK8S_Type,XlatSrcK8S_Namespace,XlatSrcK8S_Name,1) or concat(XlatSrcAddr,':',XlatSrcPort)
```

**Review by jotak — APPROVED** [review]
/lgtm

**Pull request merged**